packages/graph-explorer-proxy-server/node-server.ts

```diff
@@ -192,11 +183,11 @@ app.use(
 );
 
 // Host the Graph Explorer UI static files
-const staticFilesVirtualPath = process.env.GRAPH_EXP_ENV_ROOT_FOLDER;
```

Comment:
Is this change intentional? I thought the static files would still be located under the  extended Jupyter proxy/port path for SageMaker notebooks.

Reply:
Yeah, this bit is very confusing. But I think I've figured it out.

It breaks down like this:

- Running locally in dev we have:
    - Proxy server: `http://localhost:80`
    - Client server: `http://localhost:5173`
    - Client file root path: `/`
- Running locally in prod we have:
    - Proxy server: `http://localhost:80`
    - Client server: N/A (hosted by proxy server from `/packages/graph-explorer/dist/`)
    - Client file root path: `/explorer`
- Running on Neptune Notebook we have:
    - Proxy server: `http://localhost:9250`
    - Client server: N/A (hosted by proxy server from `/packages/graph-explorer/dist/`)
    - Client file root path: `/proxy/9250/explorer`

That last one is the tricky one. It wasn't obvious to me at first what the `GRAPH_EXP_ENV_ROOT_FOLDER` value was trying to accomplish. But here is what I think is desired.

The environment value `GRAPH_EXP_ENV_ROOT_FOLDER` is really defining the [Vite public base path](https://vitejs.dev/guide/build#public-base-path) of the client files. This is necessary so that Vite can update all the relative paths of the client files to work properly when the base path is not the root.

The proxy server, on the other hand, needs a place to host the static client files. Currently, Graph Explorer client is expected to be on the `/explorer` path. And in Neptune Notebooks, the proxy server is hosted in a virtual path of `/proxy/9250`. So those get merged to get the full virtual path for the client to use: `/proxy/9250/explorer`.

However, the proxy server does not need to know about the `/proxy/9250` part. As far as the proxy server is concerned, it is hosting the files relative to the root path on `http://localhost:9250`. It is the Jupyter server proxy that adds the virtual path `/proxy/9250`.

This is why the Express code uses the virtual path of `/explorer`. It will resolve to `http://localhost:9250/explorer` on a Notebook. Then Jupyter server will proxy that to be `http://[public-address]/proxy/9250/explorer`.

Reply:
And here is the commit where I broke the old logic.

[packages/graph-explorer-proxy-server/node-server.ts](https://github.com/aws/graph-explorer/pull/500/commits/1642e82202e18c55a3f8c6217d7507e6ad038d90)

The old logic really threw me off. What it is actually saying is:

- If `NEPTUNE_NOTEBOOK` is true, then use `/explorer`
- If `NEPTUNE_NOTEBOOK` is false and `GRAPH_EXP_ENV_ROOT_FOLDER` has a value, then use `GRAPH_EXP_ENV_ROOT_FOLDER`
- If `NEPTUNE_NOTEBOOK` is false and `GRAPH_EXP_ENV_ROOT_FOLDER` is undefined, then use `/explorer`

I believe that middle case is invalid. It would only work in cases where the proxy server is hosted at the root path, because `GRAPH_EXP_ENV_ROOT_FOLDER` must be the full public path to the client files for the client side to work at all. If the proxy server is hosted on any other path then it would break.